modules/tinymce/src/core/main/ts/SelectionOverrides.ts

```diff
@@ -132,22 +131,26 @@ const SelectionOverrides = (editor: Editor): SelectionOverrides => {
     editor.on('ResizeWindow FullscreenStateChanged', fakeCaret.reposition);
 
     const hasNormalCaretPosition = (elm: Element) => {
-      const caretWalker = CaretWalker(elm);
-
-      if (!elm.firstChild) {
+      const start = elm.firstChild;
+      if (Type.isNullable(start)) {
         return false;
       }
 
-      const startPos = CaretPosition.before(elm.firstChild);
-      const newPos = caretWalker.next(startPos);
-
-      return newPos && !isNearFakeSelectionElement(newPos);
+      const startPos = CaretPosition.before(start);
+      // If the element has a single br as a child (e.g. is empty), then the start position is a valid cursor position
+      if (NodeType.isBr(startPos.getNode()) && elm.childNodes.length === 1) {
+        return !isNearFakeSelectionElement(startPos);
+      } else {
+        const caretWalker = CaretWalker(elm);
+        const newPos = caretWalker.next(startPos);
+        return newPos && !isNearFakeSelectionElement(newPos);
+      }
     };
 
     const isInSameBlock = (node1: Node, node2: Node) => {
       const block1 = dom.getParent(node1, isBlock);
       const block2 = dom.getParent(node2, isBlock);
-      return block1 === block2;
+      return block1 === block2 || dom.isChildOf(block2, block1);
```

Comment:
Is there a reason we don't also have `|| dom.isChildOf(block1, block2)`?

Reply:
We don't want that in this case, as `node1` is the `target` element and `node2` is the `caret` element. So this should not return true if `target` is within the `caret` block element. To be honest, as this function is only used for the one purpose we probably should just rename `node1`/`block1` and `node2`/`block2.

Reply:
Actually, I'll just move the `isChildOf` check over to `hasBetterMouseTarget`. That should avoid confusion better than it being here.

Reply:
Done